.github/workflows/build_and_test.yml

```
@@ -45,6 +45,27 @@ jobs:
             run cancel-previous-in-concurrency-group.yml \
               --field concurrency_group="${{ env.E2E_CONCURRENCY_GROUP }}"
 
+  files-changed:
+    needs: [ check-permissions ]
+    name: Detect what files changed
+    runs-on: ubuntu-22.04
+    timeout-minutes: 3
+    outputs:
+      check-rust-dependencies: ${{ steps.files-changed.outputs.rust_dependencies }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Check for Postgres changes
+        uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36  # v3.0.2
+        id: files-changed
+        with:
+          token: ${{ github.token }}
```

Afair, we mostly use `${{ secrets.GITHUB_TOKEN }}` in this repo, let's continue with that

```suggestion
          token: ${{ secrets.GITHUB_TOKEN }}
```

---

it's not quite consistent. Quite a few actions default to `github.token` under the hood, and it's mostly `secrets.CI_ACCESS_TOKEN`, not `secrets.GITHUB_TOKEN`. I think going for the tokens provided by github whereever possible makes most sense, as we can scope permissions there per job, where CI_ACCESS_TOKEN has permissions defined globally

---

What I meant is that `${{ secrets.GITHUB_TOKEN }}` is the same as `${{ github.token }}`: https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication#about-the-github_token-secret

But I prefer the latter one, as it's more explicit that it's a secret 😄

---

ah, now I understand. Sure, that makes sense :)